omegaup/validator/parsing.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+"""Parsing helpers
+"""
+
+import parsy as p   # type: ignore
+
+from typing import Any, Optional
+
+space = p.string(" ").desc("space")
+eol = p.string("\n").desc("eol")
+integer = p.regex("-?[0-9]+").map(int).desc("integer")
+
+
+def boundInteger(min: Optional[int], max: Optional[int]) -> Any:
+    @p.generate(f"integer between {min} and {max}")  # type: ignore
+    def parser() -> Any:
+        num = yield integer
+        if min is not None and not min <= num:
+            return p.fail('out of bounds')
```

Comment:
parsy muestra en el error los bounds y el valor del número? si no, se puede poner aquí para mejor diagnóstico?

Reply:
Muestra el outermost label que ha recibido el parser hasta ese punto, entonces en realidad ese string de `'out of bounds'` lo tira y el diagnostic message de un usuario intentando parsear con `boundedInteger` nomás dice `expected "integer between {min} and {max} at line:col"` (otro ejemplo: a un usuario de `natural` le va a decir el error `expected "natural number" at line:col`)

Creo que ese es un problema en general no-soluble sin cosas feas de un parser combinator en python, porque no hay como una forma bonita de stackear error messages en las APIs que he visto :c

Reply:
Con tantito plumbing en omegaUp-deploy, al menos podemos apuntar directamente a la línea/columna donde explotó: eso seguro ayuda a debuggear

Reply:
vale, con eso basta entonces

Reply:
Retiro mi comentario, sí hay un `aggregate` por ahí: le investigo para que queden bonitos los logs